process/types.go

```diff
@@ -126,6 +139,10 @@ type Process interface {
 	// or deleted.
 	ExtractAsFile(string) (string, error)
```

Comment:
The `ExtractAsFile` should be removed if this is done. It is completely redundant, and now not even used (it was used for a little bit by the Rust based go symbolizer).

I also wonder if the API should be simplified by removing the `OpenMappingFile`, `GetMappingFileLastModified` and `CalculateMappingFileID` also. This can be then done with a helper if `OpenFile` is done. I suspect the only rare race condition if a file is deleted, and a file with same name is then again added. Perhaps these could be then replaced with `OpenMappingFile` that takes the Mapping and returns also `ProcessFile`?